cfg/qt.cfg

```diff
@@ -3568,33 +3594,38 @@
       <function name="back" yields="item"/>
       <function name="last" yields="item"/>
       <function name="data" yields="buffer"/>
-      <function name="shrink_to_fit" action="change-internal"/>
-      <function name="squeeze" action="change-internal"/>
-      <function name="reserve" action="change-internal"/>
+      <function name="rbegin" yields="start-iterator"/>
+      <function name="crbegin" yields="start-iterator"/>
+      <function name="rend" yields="end-iterator"/>
+      <function name="crend" yields="end-iterator"/>
     </access>
   </container>
-  <container id="qtString" startPattern="QString" endPattern="" inherits="qtContainer" opLessAllowed="true">
+  <container id="qtString" startPattern="QString" endPattern="" inherits="qtContainer" opLessAllowed="true" itEndPattern=":: iterator|const_iterator|reverse_iterator|const_reverse_iterator">
     <type string="std-like"/>
     <size>
       <function name="isNull" yields="empty"/>
       <function name="resize" action="resize"/>
+      <function name="push_back" action="push"/>
+      <function name="push_front" action="push"/>
+      <function name="prepend" action="push"/>
+      <function name="shrink_to_fit" action="change-internal"/>
+      <function name="squeeze" action="change-internal"/>
+      <function name="reserve" action="change-internal"/>
     </size>
     <access indexOperator="array-like">
       <function name="at" yields="at_index"/>
       <function name="front" yields="item"/>
       <function name="back" yields="item"/>
+      <function name="rbegin" yields="start-iterator"/>
+      <function name="crbegin" yields="start-iterator"/>
+      <function name="rend" yields="end-iterator"/>
+      <function name="crend" yields="end-iterator"/>
     </access>
   </container>
-  <container id="qtStringList" startPattern="QStringList" inherits="qtContainer" opLessAllowed="true">
-    <type string="std-like"/>
```

Comment:
@danmar I am not sure if it is correct to remove this. Without this type specification `QStringList` inherits from `QList` which inherits `<type templateParameter="0"/>` from the qtContainer configuration. This is also wrong.
Should I just add it back? For the tests that I have added it does not make any difference.

And BTW it also does not any difference how the itEndPattern is specified.

Reply:
@danmar Any opinion on that?

Reply:
I guess removing an inherited templateParameter is currently not implemented. We could add removal via a value of `-1` or so for templateParameter.